authors/apps/articles/models.py

```diff
@@ -3,8 +3,47 @@
     generate_slug, unique_code_generator,
     get_likes_or_dislkes, get_average_value)
 from django.db.models.signals import pre_save
+from authors.apps.authentication.models import User 
 from authors.apps.profiles.models import Profile
-from authors.apps.authentication.models import User
+from authors.apps.utils.messages import error_messages, favorite_actions_messages
+
+
+class ArticleManager(models.Manager):
+    def handle_favorite_actions(self, request_user_obj, article_slug):
+        """ 
+        Enales both favoriting of articles 
+        Args:
+            request_user_obj:  Django http request user
+            article_slug: Article model instance
+
+        """
+        profile_ = Profile.objects.filter(user=request_user_obj).first()
+        article_ = self.model.objects.filter(slug=article_slug).first()
+        article_.favoritesCount = article_.favorites.count() + 1
+        article_.favorited = True
+        article_.favorites.add(profile_)
+        article_.save()
+        return favorite_actions_messages.get('favorited')
+
+    def handle_unfavorite(self, **kwargs):
+        """ 
+        Handles unfavoriting of articles 
+        Args:
+            request_user:  currently loggedin user
+            article_slug: Article model instance
+
+        """
+        article_slug = kwargs.get("article_slug")
+        request_user = kwargs.get("request_user")
+        user_ = Profile.objects.get(user=request_user)
+        article_to_unfavorite = self.model.objects.get(slug=article_slug)
+        if article_to_unfavorite.favorites.count() > 0:
```

Comment:
@armstrongsouljah Have tried using
```suggestion
        if article_to_unfavorite.favorites.count():
```

Reply:
No i haven't. Let me try and see if it renders the same output, I will change it to that.

Reply:
It actually works the same way. Let me push the changes.